redbot/cogs/warnings/warnings.py

```diff
@@ -301,19 +346,43 @@ def __init__(self, bot: Red):
 
         await warning_points_add_check(self.config, ctx, user, current_point_count)
         try:
+            dm = await self.config.guild(ctx.guild).toggle_dm()
+            if dm == True:
+                em = discord.Embed(
+                    title=_("Warning from {user}").format(user=ctx.author),
+                    description=reason_type["description"],
+                )
+                em.add_field(name=_("Points"), value=str(reason_type["points"]))
+                await user.send(
+                    _("You have received a warning in {guild_name}.").format(
+                        guild_name=ctx.guild.name
+                    ),
+                    embed=em,
+                )
+                await ctx.send(_("User {user} has been warned in their DM's.").format(user=user))
+        except discord.HTTPException:
+            await ctx.send(
+                "The warning couldn't be sent to the user because the user has DM's from server members turned off."
+            )
+        toggle_channel = await self.config.guild(ctx.guild).toggle_channel()
+        if toggle_channel == True:
             em = discord.Embed(
                 title=_("Warning from {user}").format(user=ctx.author),
                 description=reason_type["description"],
             )
             em.add_field(name=_("Points"), value=str(reason_type["points"]))
-            await user.send(
-                _("You have received a warning in {guild_name}.").format(
-                    guild_name=ctx.guild.name
-                ),
-                embed=em,
+            warn_channel = await self.config.guild(ctx.guild).warn_channel()
+            if warn_channel == None:
+                channel = ctx.channel
+            else:
+                channel == warn_channel
+                await channel.send(
+                    _("{user} has been warned.").format(user=user.mention), embed=em
+                )
+        if dm and toggle_channel == False:
```

Comment:
I would remove this whole section and go back to how it was handled in this line, because there's no need to inform the user each time that DM and channel warns are toggled off:
https://github.com/Cog-Creators/Red-DiscordBot/pull/2814/files#diff-bd66023271557007376afcf648f3b3ffL341

Reply:
I feel like it should as there is quite a big chance that not every moderator in a server would know exactly how the bot is set up and may not be familiar with the bot and what it means if it says nothing after a warn has been given.